src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -532,21 +539,25 @@ private void Poll()
         private List<ConsumeResult<K, V>> PollKafka(CancellationToken token)
         {
             ConsumeResult<K, V> consumed = null;
-            var timeout = (int) _pollTimeout.TotalMilliseconds;
+            var i = 10; // 10 poll attempts
+            var timeout = (int) _pollTimeout.TotalMilliseconds / i;
```

Comment:
Might need to add a check here to make sure that this value is never 0. Casting to int always rounds down, so the timeout can be 0. A `Math,Max()` call should be enough.

Reply:
Polling 0 returns immediately with whatever is stored in the buffer, i think. Let me check.

Reply:
Yeah, it should returns a single entry from the internal poll buffer or null of there's none.

Reply:
I wonder if we'd be better off just waiting the entire time, rather than polling in a loop.

Reply:
Ah no, we're not - could only poll a maximum of 1 messages if we did that.

Reply:
Fixed